hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/opt/physical/visitor/RexToExpression.java

```diff
@@ -371,8 +381,6 @@ private RexToExpression() {
                 break;
 
             case BIGINT:
-                // XXX: Calcite returns unscaled value of the internally stored
```

Comment:
Looks like the comment still holds.

Reply:
Is it a problem for us? BIGINT type is resolved only when the value is within the value range of the Long type, and there are no decimal points. Therefore, unscaling should not affect us anyhow, unless I am missing something. 
In any case, I simplified this code to just request the `Long.class` value, it should work well either: https://github.com/hazelcast/hazelcast/pull/17947/commits/65df331cf8df7c9690ee4f961f2a1d81a905fdfc

Reply:
`select cast(1.1 as bigint)` now returns `11`. There is obviously some gaps in the test coverage, at least for `CAST`, since the latest build didn't fail.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/17947/commits/7e28f0c1d4779631a20f8ea26c335c307137971a, missing test is added